app/services/dashboard/recent_work_service.rb

```diff
@@ -0,0 +1,164 @@
+# frozen_string_literal: true
+
+module Dashboard
+  class RecentWorkService
+    include InputSanitizeHelper
+    include Rails.application.routes.url_helpers
+
+    def initialize(user, team, mode)
+      @user = user
+      @team = team
+      @mode = mode
+    end
+
+    def call
+      visible_projects = Project.viewable_by_user(@user, @team)
+      visible_by_team = activities_with_filter.where(project: nil, team_id: @team.id)
+      visible_by_projects = activities_with_filter.where(project_id: visible_projects.pluck(:id))
+      query = Activity.from("((#{visible_by_team.to_sql}) UNION ALL (#{visible_by_projects.to_sql})) AS activities")
+
+      # Join subjects
+      query = query.joins("
+                        LEFT JOIN results ON
+                          subject_type = 'Result'
+                          AND subject_id = results.id
+                        LEFT JOIN protocols ON
+                          subject_type = 'Protocol'
+                          AND subject_id = protocols.id
+                        LEFT JOIN my_modules ON
+                          (subject_type = 'MyModule' AND subject_id = my_modules.id)
+                          OR  protocols.my_module_id = my_modules.id
+                          OR  results.my_module_id = my_modules.id
+                        LEFT JOIN experiments ON
+                          (subject_type = 'Experiment' AND subject_id = experiments.id)
+                          OR experiments.id = my_modules.experiment_id
+                        LEFT JOIN projects ON
+                          (subject_type = 'Project' AND subject_id = projects.id)
+                          OR projects.id = experiments.project_id
+                        LEFT JOIN repositories ON
+                          subject_type = 'Repository'
+                          AND subject_id = repositories.id
+                        LEFT JOIN reports ON subject_type = 'Report'
+                          AND subject_id = reports.id
+                      ")
+                   .where("
+                        (projects.archived != 'true' OR projects.archived IS NULL)
+                        AND (experiments.archived != 'true' OR experiments.archived IS NULL)
+                        AND (my_modules.archived != 'true' OR my_modules.archived IS NULL)
+                        AND (protocols.protocol_type != 4 OR protocols.protocol_type IS NULL)
```

Comment:
All these are NOT NULL

Reply:
query = query.where('projects.archived IS NOT TRUE')
                   .where('experiments.archived IS NOT TRUE')
                   .where('my_modules.archived IS NOT TRUE')
                   .where('protocols.protocol_type != ? OR protocols.protocol_type IS NULL',
                          Protocol.protocol_types[:in_repository_archived])